audiofile/core/io.py

```diff
@@ -386,7 +387,11 @@ def read(
                 offset /= sampling_rate
             if duration is not None and duration != 0:
                 duration /= sampling_rate
-            convert(file, tmpfile, offset, duration)
+            if sampling_rate is None:
```

Comment:
This is the essential bit then:

for opus conversion, you are forcing convert to have the sampling rate parameter. 
Should one add a comment why this is needed?